listeners/FilterTimeslot.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Igniter\Local\Listeners;
+
+use Admin\Models\Orders_model;
+use Carbon\Carbon;
+use DateTime;
+use Igniter\Flame\Location\Models\AbstractLocation;
+use Igniter\Local\Classes\Location;
+use Igniter\Local\Facades\Location as LocationFacade;
+use Igniter\Flame\Traits\EventEmitter;
+use Illuminate\Contracts\Events\Dispatcher;
+
+class FilterTimeslot
+{
+    use EventEmitter;
+
+    protected static $ordersCache = [];
+    protected static $scheduleCache = [];
+
+    public function subscribe(Dispatcher $dispatcher)
+    {
+        $dispatcher->listen('igniter.workingSchedule.timeslotFilter', __CLASS__.'@filterTimeslot');
+    }
+
+    public function filterTimeslot($workingSchedule, $timeslot)
+    {
+        // Skip if the working schedule is not for delivery or pickup
+        if ($workingSchedule->getType() == AbstractLocation::OPENING)
+            return;
+            
+        // allow the timeslot selection logic to be overwritten by extensions
+        if ($event = $this->fireSystemEvent('igniter.local.timeslotValid', [$timeslot])) {
+            return $event;
+        }
+            
+        $dateString = Carbon::parse($timeslot)->toDateString();
+            
+        $ordersOnThisDay = $this->getOrders($dateString);
+                
+        $location = new Location(LocationFacade::getId());
+        $limitTimeslots = $this->getSchedule($location, $dateString);
+
+        foreach ($limitTimeslots as $limitDate => $limitHoursArray)
```

Comment:
> https://github.com/tastyigniter/TastyIgniter/pull/447#discussion_r470572291

If you use the delivery/pickup interval/lead time instead of the new limit_orders_interval here, then all you'll need to do is create a start and end time using the delivery/pickup interval/lead time then count the total number of orders between the start and end time and check if it is less than limit_orders_count. No need for the $limitTimeslots.

Reply:
I get you - like a rolling time interval rather than a set one.
I've updated it.